### PR TITLE
add dist branch for add_n op.

### PR DESCRIPTION
**Pull request by winter-wang**
### PR Category

Auto Parallel

### PR Types

New features

### Description

- add dist branch for add_n op.

### Other
Pcard-67164

### CODE REVIEW

**Review by zhiqiu — APPROVED** [review]
LGTM

File: paddle/fluid/pir/dialect/operator/ir/manual_op.cc [hunk 1]
```diff
@@ -190,14 +200,36 @@ std::vector<pir::Type> AddNOp::InferMeta(
...
-      TransToIrDataType(dense_out.dtype()),
-      dense_out.dims(),
-      dense_out.layout(),
-      dense_out.lod(),
-      dense_out.offset());
-  argument_outputs.push_back(out_dense_tensor_type);
+  pir::Type out_type = CvtToDenseTensorType(dense_out);
+#ifdef PADDLE_WITH_DISTRIBUTE
+  // Auto Parallel condition
+  if (AllInputAreDist(input_values)) {
```
Comment (JZ-LIANG) [inline paddle/fluid/pir/dialect/operator/ir/manual_op.cc:206]: Raise Error if part of inputs are dist

**Review by JZ-LIANG — APPROVED** [review]
LGTM

File: paddle/fluid/pir/dialect/operator/ir/manual_op.cc (hunk 1 above)
Comment (winter-wang) [reply paddle/fluid/pir/dialect/operator/ir/manual_op.cc:206]: Thanks!, I will fix it in the next pr.

**Pull request merged**